src/main/java/net/minecraftforge/common/extensions/IForgeDispensibleContainerItem.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.minecraftforge.common.extensions;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.DispensibleContainerItem;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.phys.BlockHitResult;
+import org.jetbrains.annotations.Nullable;
+
+public interface IForgeDispensibleContainerItem
+{
+    default boolean emptyContents(@Nullable Player player, Level level, BlockPos blockPos, @Nullable BlockHitResult blockHitResult, @Nullable ItemStack container)
```

Comment:
`BlockPos` can be shortened to `pos` and `BlockHitResult` to `hitResult`. Stating the class name is not the best solution. Also, this is missing documentation.

Reply:
reworked